### ISSUE TITLE
realm trigger causes provider to authenticate with atlas even if realm triggers are not in use

### ISSUE THREAD

**Issue by jooh-lee** [mongodb/terraform-provider-mongodbatlas#512]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
terraform v0.14
provider version 1.0
```

### Terraform Configuration File 

```hcl
We are not using any mongo features in this particular tf workspace, but a common module is causing the provider to be downloaded
```

### Steps to Reproduce

1. Have the mongo provider downloaded via init
2. Do not set the api keys
3. Do not use any mongo resources
4. Plan

### Expected Behavior

when the feature is not in use, it should not be trying to contact atlas realm function to fetch info

### Actual Behavior

Even if no mongo resources are in use, the provider tries to authenticate causing the below error:
`Error: Post "https://realm.mongodb.com/api/admin/v3.0/auth/providers/mongodb-cloud/login": challenge is bad`

This was not a problem before version 1. We are currently on v0.9 and when no mongo resources are being used the provider doesnt try to authenticate with atlas.

**Comment by themantissa** [comment]
@jooh-lee thank you so much for the report.  We will have our team take a look at this.

**Comment by themantissa** [comment]
Internal ticket: INTMDB-242

**Comment by themantissa** [comment]
We have a pre-release of 1.0.1 ready - we'll release the GA version tomorrow.  If you have time to try it out before then it's here: https://github.com/mongodb/terraform-provider-mongodbatlas/releases/tag/v1.0.1-pre.1

**Comment by themantissa** [comment]
Fixed in recent release 1.0.1
